website/content/en/docs/kubebuilder/advanced-topics.md

```diff
@@ -138,61 +138,82 @@ import (
 
 const memcachedFinalizer = "finalizer.cache.example.com"
 
-func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
-	reqLogger.Info("Reconciling Memcached")
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("memcached", req.NamespacedName)
 
 	// Fetch the Memcached instance
 	memcached := &cachev1alpha1.Memcached{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+	err := r.Get(ctx, req.NamespacedName, memcached)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			reqLogger.Info("Memcached resource not found. Ignoring since object must be deleted.")
-			return reconcile.Result{}, nil
+			log.Info("Memcached resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		reqLogger.Error(err, "Failed to get Memcached.")
-		return reconcile.Result{}, err
+		log.Error(err, "Failed to get Memcached")
+		return ctrl.Result{}, err
 	}
 
-	...
-
-	// Check if the Memcached instance is marked to be deleted, which is
-	// indicated by the deletion timestamp being set.
-	isMemcachedMarkedToBeDeleted := memcached.GetDeletionTimestamp() != nil
-	if isMemcachedMarkedToBeDeleted {
-		if contains(memcached.GetFinalizers(), memcachedFinalizer) {
-			// Run finalization logic for memcachedFinalizer. If the
-			// finalization logic fails, don't remove the finalizer so
-			// that we can retry during the next reconciliation.
-			if err := r.finalizeMemcached(reqLogger, memcached); err != nil {
-				return reconcile.Result{}, err
-			}
-
-			// Remove memcachedFinalizer. Once all finalizers have been
-			// removed, the object will be deleted.
-			controllerutil.RemoveFinalizer(memcached, memcachedFinalizer)
-			err := r.client.Update(context.TODO(), memcached)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
+	// Check if the deployment already exists, if not create a new one
```

Comment:
@theishshah You've removed the use of the finalizer code `finalizeMemcached()` from this example altogether.
I think my goal for this was just to update the code's variable names and signatures according to the new scaffold.

Right now this example is not showing how to use finalizers at all.
Please revert the old code example but just update the vars and function signatures.